google-cloud-spanner/lib/google/cloud/spanner/session.rb

```diff
@@ -338,13 +343,11 @@ def path
         #
         def execute_query sql, params: nil, types: nil, transaction: nil,
                           partition_token: nil, seqno: nil, query_options: nil,
-                          request_options: nil, call_options: nil, data_boost_enabled: nil
+                          request_options: nil, call_options: nil, data_boost_enabled: nil,
+                          directed_read_options: nil
           ensure_service!
-          if query_options.nil?
-            query_options = @query_options
-          else
-            query_options = @query_options.merge query_options unless @query_options.nil?
-          end
+          query_options = merge_if_present query_options, @query_options
+          directed_read_options = merge_if_present directed_read_options, @directed_read_options
```

Comment:
Where are we considering client level directed_read_options when request level is nil?

Reply:
and how are we ensuring that client level option is set only for Single and Readonlytransactions?

Reply:
I have updated the options to flow from clients directly rather than doing it at session which makes it much more cleaner and the precedence for option is handled at client apis itself.

Reply:
Makes sense